vms/avm/utxo/spender.go

```diff
@@ -5,46 +5,20 @@ package utxo
 
 import (
 	"errors"
-	"fmt"
 
 	"github.com/ava-labs/avalanchego/codec"
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/utils/crypto/secp256k1"
-	"github.com/ava-labs/avalanchego/utils/math"
 	"github.com/ava-labs/avalanchego/utils/timer/mockable"
 	"github.com/ava-labs/avalanchego/vms/avm/txs"
 	"github.com/ava-labs/avalanchego/vms/components/avax"
 	"github.com/ava-labs/avalanchego/vms/nftfx"
 	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
 )
 
-var (
-	errSpendOverflow          = errors.New("spent amount overflows uint64")
-	errInsufficientFunds      = errors.New("insufficient funds")
-	errAddressesCantMintAsset = errors.New("provided addresses don't have the authority to mint the provided asset")
-)
+var errInsufficientFunds = errors.New("insufficient funds")
 
 type Spender interface {
-	// Spend the provided amount while deducting the provided fee.
-	// Arguments:
-	// - [utxos] contains assets ID and amount to be spend for each assestID
-	// - [kc] are the owners of the funds
-	// - [amounts] is the amount of funds that are available to be spent for each assetID
-	// Returns:
-	// - [amountsSpent] the amount of funds that are spent
-	// - [inputs] the inputs that should be consumed to fund the outputs
-	// - [signers] the proof of ownership of the funds being moved
-	Spend(
-		utxos []*avax.UTXO,
-		kc *secp256k1fx.Keychain,
-		amounts map[ids.ID]uint64,
-	) (
-		map[ids.ID]uint64, // amountsSpent
-		[]*avax.TransferableInput, // inputs
-		[][]*secp256k1.PrivateKey, // signers
-		error,
-	)
-
 	SpendNFT(
```

Comment:
I didn't move this to the wallet to keep change as small as possible. Given the recent, deeper cleanup in the P-chain, we may consider doing it so to remove spender completely